catalog/src/main/java/it/eng/catalog/service/CatalogService.java

```diff
@@ -3,24 +3,36 @@
 import java.util.List;
 import java.util.Optional;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.stereotype.Service;
 
 import it.eng.catalog.exceptions.CatalogErrorException;
 import it.eng.catalog.model.Catalog;
 import it.eng.catalog.model.Dataset;
+import it.eng.catalog.model.Offer;
+import it.eng.catalog.model.Serializer;
 import it.eng.catalog.repository.CatalogRepository;
+import it.eng.tools.event.contractnegotiation.ContractNegotationOfferRequest;
+import it.eng.tools.event.contractnegotiation.ContractNegotiationOfferResponse;
+import jakarta.validation.ValidationException;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * The CatalogService class provides methods to interact with catalog data, including saving, retrieving, and deleting catalogs.
  */
 @Service
+@Slf4j
 public class CatalogService {
 
-    private final CatalogRepository repository;
+	@Autowired
+    private CatalogRepository repository;
+	@Autowired
+    private ApplicationEventPublisher publisher;
 
-    public CatalogService(CatalogRepository repository) {
-        this.repository = repository;
-    }
+//    public CatalogService(CatalogRepository repository) {
```

Comment:
Any specific reason for avoiding Constructor Dependency Injection?

Reply:
I saw somewhere that they are suggesting field injection, but did not save article.
After googleing and checking Spring official documentation, they are suggesting/proposing constructor injection. so I guess, I should revert it back